scripts/dist-git/remove_unused_sources

```diff
@@ -0,0 +1,75 @@
+#!/bin/bash
+
+Usage() {
+    cat <<EOF
+Usage:
+    $0 <package_git_directory> <package_lookaside_directory>
+
+    Removes all tarballs in <package_lookaside_directory> except
+    for tarballs that are referenced by the latest commit of each
+    branch in <package_git_directory>.
+EOF
+}
+
+if [[ $# != 2 ]]; then
+    Usage
+    exit 1
+fi
+
+pkg_git_dir="$1"
+pkg_lookaside_dir="$2"
+
+if [ ! -d "$pkg_git_dir" ]; then
+    echo "$pkg_git_dir is not a valid directory."
+    exit 1
+fi
+if [ ! -d "$pkg_lookaside_dir" ]; then
+    echo "$pkg_lookaside_dir is not a valid directory."
+    exit 1
+fi
+
+pushd "$pkg_git_dir" > /dev/null
+
+whitelist=()
+
+# find sources that are referenced by the latest commit in any of the branches
+for branch in `git for-each-ref --format="%(refname:short)" refs/heads`; do
+    while read line; do
+        set -- $line
+        hash=$1
+        filename=$2
+        # skip projects using the new format
+        test $# = 2 -a $filename = `echo "$filename" | sed -e "s/^(//" -e "s/)$//"` || exit 0
+        whitelist+=("$hash","$filename")
+    done < <(git show "$branch":sources)
+done
+
+popd > /dev/null
+
+# remove all source files that are not referenced
+while read file; do
+    old_IFS=$IFS
+    IFS=/
+    set -- $file
+
+    # safety measure, if this is really the layout we expect the first and
+    # third part matches
+    test "$1" = "$3" || continue
+
+    keep=false
+    for source in "${whitelist[@]}"; do
+        IFS=','
+        read -ra parsedSource <<< "$source"
```

Comment:
I'd still prefer `set -- $source` here.  It would override your `$1` and `$2`, but it's even better to do
```bash
filename=$1
hash=$2
```
... above, and do again whatever we want with `$@`.